seed/csharp-model/trace/src/SeedTrace/V2/V3/Problem/TestCaseImplementationDescriptionBoard.cs

```diff
@@ -182,14 +182,14 @@ JsonSerializerOptions options
                 discriminatorElement.GetString()
                 ?? throw new JsonException("Discriminator property 'type' is null");
 
-            var value = discriminator switch
-            {
-                "html" => json.GetProperty("value").Deserialize<string>(options)
-                    ?? throw new JsonException("Failed to deserialize string"),
-                "paramId" => json.GetProperty("value").Deserialize<string>(options)
-                    ?? throw new JsonException("Failed to deserialize string"),
-                _ => json.Deserialize<object?>(options),
-            };
+            var value =
+                discriminator switch
+                {
+                    "html" => json.GetProperty("value").Deserialize<string>(options),
+                    "paramId" => json.GetProperty("value").Deserialize<string>(options),
+                    _ => json.Deserialize<object?>(options),
```

Comment:
This worries me, some discriminants are nullable, so we shouldn't throw on null. Moving the `?? throw` to the bottom will be a breaking change/bug, unless we calculate up front that none of the discriminants can be null.

Reply:
Yeah, I didn't catch that case. My Bad.

HMMMMM.

This is somewhat tricky. I **_can_** check if the individual ones are optional, but there isn't detail to identify when the type is guaranteed **_not_** to be null (in which case the compiler complains that the `?? throw ...` isn't permitted. (really, the compiler complaining about the `??` because it knows it isn't going to happen is pretty petty on it's part. 

I can check to see if any are optional and suppress the `?? ` entirely in that case since, that would be a valid result regardless of what's in the switch

Reply:
- If the type from IR is optional or nullable, we should not `?? throw`
- If the type from IR is primitive, we should not `?? throw`
- Else `?? throw`

I think we have some code for determining whether something is `primitive`; it's not the most reliable, but good enough.

Reply:
It appears to respect nullable/optional types, but not the primitive case.